dlt/common/runtime/segment.py

```diff
@@ -23,24 +23,30 @@
 _SESSION: requests.Session = None
 _WRITE_KEY: str = None
 _SEGMENT_REQUEST_TIMEOUT = (1.0, 1.0)  # short connect & send timeouts
-_SEGMENT_ENDPOINT = "https://api.segment.io/v1/track"
+_SEGMENT_ENDPOINT: str = None
 _SEGMENT_CONTEXT: TExecutionContext = None
 
 
 def init_segment(config: RunConfiguration) -> None:
-    assert (
-        config.dlthub_telemetry_segment_write_key
-    ), "dlthub_telemetry_segment_write_key not present in RunConfiguration"
+    global _WRITE_KEY, _SESSION, _SEGMENT_ENDPOINT
+    if "RUNTIME__TELEMETRY_ENDPOINT" not in os.environ:
```

Comment:
you do not need to access os.environ, accessing the values in the config is enough. what you need to do here:
* if config.dlthub_telemetry_endpoint is "https://api.segment.io/v1/track", config.dlthub_telemetry_segment_write_key should be provided, if not fail
* if config.dlthub_telemetry_endpoint is None, fail
* if config.dlthub_telemetry_endpoint is something else, it's ok

Reply:
fyi, your environ var does not match the config, so maybe that is the reason for your confusion.